config/src/test/scala/com/typesafe/config/impl/TokenizerTest.scala

```diff
@@ -239,15 +286,20 @@ class TokenizerTest extends TestUtils {
     def commentsHandledInVariousContexts() {
         tokenizerTest(List(tokenString("//bar")), "\"//bar\"")
         tokenizerTest(List(tokenString("#bar")), "\"#bar\"")
-        tokenizerTest(List(tokenUnquoted("bar"), tokenComment("comment")), "bar//comment")
-        tokenizerTest(List(tokenUnquoted("bar"), tokenComment("comment")), "bar#comment")
-        tokenizerTest(List(tokenInt(10), tokenComment("comment")), "10//comment")
-        tokenizerTest(List(tokenInt(10), tokenComment("comment")), "10#comment")
-        tokenizerTest(List(tokenDouble(3.14), tokenComment("comment")), "3.14//comment")
-        tokenizerTest(List(tokenDouble(3.14), tokenComment("comment")), "3.14#comment")
+        tokenizerTest(List(tokenUnquoted("bar"), tokenComment("comment", true)), "bar//comment")
+        tokenizerTest(List(tokenUnquoted("bar"), tokenComment("comment", false)), "bar#comment")
+        tokenizerTest(List(tokenInt(10), tokenComment("comment", true)), "10//comment")
+        tokenizerTest(List(tokenInt(10), tokenComment("comment", false)), "10#comment")
+        tokenizerTest(List(tokenDouble(3.14), tokenComment("comment", true)), "3.14//comment")
+        tokenizerTest(List(tokenDouble(3.14), tokenComment("comment", false)), "3.14#comment")
         // be sure we keep the newline
-        tokenizerTest(List(tokenInt(10), tokenComment("comment"), tokenLine(1), tokenInt(12)), "10//comment\n12")
-        tokenizerTest(List(tokenInt(10), tokenComment("comment"), tokenLine(1), tokenInt(12)), "10#comment\n12")
+        tokenizerTest(List(tokenInt(10), tokenComment("comment", true), tokenLine(1), tokenInt(12)), "10//comment\n12")
+        tokenizerTest(List(tokenInt(10), tokenComment("comment", false), tokenLine(1), tokenInt(12)), "10#comment\n12")
+        // be sure we handle multi-line comments
+        tokenizerTest(List(tokenComment("comment", true), tokenLine(1), tokenComment("comment2", true)),
+                      "//comment\n//comment2")
+        tokenizerTest(List(tokenComment("comment", false), tokenLine(1), tokenComment("comment2", false)),
+                      "#comment\n#comment2")
```

Comment:
could torture this a little more, like several lines of comments with leading and trailing whitespace, maybe a `\r\n`, etc.